File: 2529. Maximum Count of Positive Integer and Negative Integer/Solution.cpp
#include <ranges>
#include <vector>
using namespace std;

class Solution {
public:
    int maximumCount(vector<int>& nums)
    {
        int neg_cnt = ranges::lower_bound(nums, 0) - nums.begin();
        int pos_cnt = nums.end() - ranges::upper_bound(nums, 0);
        return max(pos_cnt, neg_cnt);
    }
};

File: 2529. Maximum Count of Positive Integer and Negative Integer/Solution.py
import bisect
from typing import List


class Solution:
    def maximumCount(self, nums: List[int]) -> int:
        neg_cnt = bisect.bisect_left(nums, 0)
        pos_cnt = len(nums) - bisect.bisect_right(nums, 0)
        return max(neg_cnt, pos_cnt)
